File: EscapeRoom/src/gestorAplicacion/clasesLogicas/Armas.java
package gestorAplicacion.clasesLogicas;
import java.io.Serializable;


import java.util.ArrayList;

public class Armas implements Herramientas, Serializable {
    private String description = "objeto generico";
    private final String name;
    private final int bonusDamage, bonusSpeed;
    private static ArrayList<Armas> armas = new ArrayList<>();
    
    public static ArrayList<Armas> getArmas() {
		return armas;
	}

	public Armas(String name, int bdamage, int bspeed){
    	this.name = name;
        this.bonusDamage = bdamage;
        this.bonusSpeed = bspeed;
    }
    
    public Armas(String name){//para crear objetos como computadores,ventanas o cosas que no se puedan recoger.
    	this.name = name;
        this.bonusDamage = 0;
        this.bonusSpeed = 0;
    }
    
    //METODOS GET
    
    @Override
    public String getName(){
        return name;
    } 
    @Override
    public String getDescription(){
        return description;
    }
    public int getBonusDamage(){
        return bonusDamage;
    }
    public int getBonusSpeed(){
        return bonusSpeed;
    }
    
    //METODO SET
    public void setDescripion(String i){
        description = i;
    }

    @Override
    public void usar(Individuo i) {
       
    }
}

File: python/Armas.py
from herramientas import Herramientas

class Armas(Herramientas): #Revisar lo del serializado
    _armas = []

    @staticmethod
    def getArmas():
        return Armas._armas

    def __init__(self, name, bdamage, bspeed):
        self._description = "objeto generico"
        self._name = name
        self._bonusDamage = bdamage
        self._bonusSpeed = bspeed
        Armas._armas.append(self)
   
    #METODOS GET

    def getName(self):
        return self._name
    def getDescription(self):
        return self._description
    def getBonusDamage(self):
        return self._bonusDamage
    def getBonusSpeed(self):
        return self._bonusSpeed

    #METODO SET
    def setDescripion(self, i):
        self._description = i

    def usar(self, i):
        pass

    def toString(self):
        m = "nombre: " + self._name + "\ndescripcion: " + self._description + "\nbonus damage: " + str(self._bonusDamage) + "\nbonus speed: " + str(self._bonusSpeed) + "\n"
        return m
